uc-crux-llvm/src/UCCrux/LLVM/Run/Simulate.hs

```diff
@@ -536,7 +537,16 @@ mkCallbacks appCtx modCtx funCtx halloc callbacks constraints cfg llvmOpts =
         IO r) ->
       IO r
     mkResultHook sym skipOverrideRef unsoundOverrideRef explRef memRef argShapeRef cruxResult resHook =
-      do unsoundness' <-
+      do for_ (traces cruxResult) $
+           \trace ->
+             do liftIO $ (appCtx ^. log) Hi "Trace:"
+                for_ trace $
+                  \loc ->
+                    if What4.InternalPos == What4.plSourceLoc loc
+                    then pure ()
+                    else liftIO $ (appCtx ^. log) Hi ("  " <> ppProgramLoc loc)
```

Comment:
This can be simplified slightly with `Control.Monad.unless`:

```suggestion
                    unless (What4.InternalPos == What4.plSourceLoc loc) $
                      liftIO $ (appCtx ^. log) Hi ("  " <> ppProgramLoc loc)
```

Reply:
Thanks!